### PR TITLE
Adding implementation to represent traffic line with smooth gradient …

### PR DESCRIPTION
**Pull request by cafesilencio**
### Description
The traffic line displays different colored sections depending on the traffic congestion. The colors abruptly change from one color to the next. The work in this PR adds an option to display the color transition with a gradient between the colors. There is an option to enable this feature as well as an option that influences the length of the color transition.

### Changelog
```
<changelog>Add option to display a gradient between the colors representing traffic congestion on the route line.
</changelog>
```

### Screenshots or Gifs
Screenshot_20210824-105549

### CODE REVIEW

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/model/MapboxRouteLineOptions.kt [hunk 1]
```diff
@@ -107,6 +120,8 @@ class MapboxRouteLineOptions private constructor(
             "tolerance=$tolerance, " +
             "displayRestrictedRoadSections=$displayRestrictedRoadSections, " +
             "styleInactiveRouteLegsIndependently=$styleInactiveRouteLegsIndependently" +
+            "displaySoftGradientForTraffic=$displaySoftGradientForTraffic" +
+            "softGradientTransition=$softGradientTransition" +
```
Comment (Guardiola31337) [inline libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/model/MapboxRouteLineOptions.kt:124]:
```suggestion
            "styleInactiveRouteLegsIndependently=$styleInactiveRouteLegsIndependently, " +
            "displaySoftGradientForTraffic=$displaySoftGradientForTraffic, " +
            "softGradientTransition=$softGradientTransition" +
```

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/model/MapboxRouteLineOptions.kt [hunk 2]
```diff
@@ -215,10 +242,35 @@ class MapboxRouteLineOptions private constructor(
...
+        fun displaySoftGradientForTraffic(enable: Boolean): Builder =
+            apply { this.displaySoftGradientForTraffic = enable }
+
+        /**
+         * When [displayRestrictedRoadSections] is set to true this value will influence the
+         * length of the gradient transition between traffic congestion colors. Larger values
+         * will result in the color transition occurring over a longer length of the line.
+         * Values between 5 and 75 are recommended. The default value is 30.0
+         *
+         * @param transitionDistance a value above zero used for the gradient transition distance
```
Comment (Guardiola31337) [inline libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/model/MapboxRouteLineOptions.kt:268]: Do we need to check at creation if the value is above zero and return an `IllegalArgumentException` if not?

**Review by Guardiola31337 — APPROVED** [review]
Overall looks good to me @cafesilencio 

Left a couple of minor comments.

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/model/MapboxRouteLineOptions.kt (hunk 2 above)
Comment (cafesilencio) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/model/MapboxRouteLineOptions.kt:268]: Yes.

**Pull request merged**